Skills table of contents
[Table of contents](#skills-table-of-contents)
- [Personal skills](#personal-skills)
- [Science topics/disciplines](#science-topicsdisciplines)
- [Wet-lab](#wet-lab)
- [Programming](#programming)
- [Bioinformatics](#bioinformatics)
- [Statistical analysis](#statistical-analysis)
- [Writing](#writing)


# Personal skills
  * Organization
    * Lab notebook/Code repository
      * Good: Evernote, OneNote, Computer files
      * Adequate: Physical notebook
      * Best: GitHub
    * File organization
    * Software to consider:
      * Manuscript and reference management: Zotero
  * Public speaking
    * "Elevator speech"
    * Poster presentations
    * Oral presentations
# Science topics/disciplines
  * Molecular biology
  * Laboratory methods (DNA extraction, library prep, sequencing)
  * Genetics
  * Bacterial ecology
  * Antimicrobial resistance
  * Livestock production
  * Project-specific biology
  * Epidemiology
# Wet-lab
  * DNA extraction
  * Library preparation
    * Target enrichment
  * Metagenomic sequencing
# Programming
  * [Important programming concepts](https://github.com/Microbial-Ecology-Group/Bioinformatics_resources/blob/master/sections/Programming/Important_programming_concepts.md):
    * Operating systems
    * Programming languages
    * computing "Environment"
    * Software/programs/executables
    * Command line
    * Functions
    * etc.
  * Languagues
    * Main languages:
      * bash/shell: used to work in computing clusters, run bioinformatic pipelines
      * R programming: most, if not all of your statistics can be run in R
    * Optional:
      * python: considered the easiest language to learn and use.
  * Setting up your environment
    * On the server
    * Your personal computer
  * [How to look for help](https://github.com/Microbial-Ecology-Group/Bioinformatics_resources/blob/master/sections/Programming/How_to_look_for_help.md)
  * [Tips for using someone else's code](https://github.com/Microbial-Ecology-Group/Bioinformatics_resources/blob/master/sections/Programming/How_to_user_others_code.md)
# Bioinformatics
  * Using the server
  * Common file types
  * Read quality filtering
  * Host sequence removal
  * Read classification
  * Running bioinformatic pipelines
    * AMR++
    * QIIME2
    * Learning to run different pipelines
# Statistical analysis
  * Important statistical concepts
    * Sequencing count data 
    * Rarefaction
    * Count normalization
    * Diversity
    * Linear models
    * Non-parametric testing
    * Differential abundance testing
    * Bayesian vs Frequentist statistics
    * etc.
  * Data exploration
  * Data visualization
  * Statistics
    * Sequencing effort
    * Alpha diversity - Wilcox testing
    * Beta diversity - Ordination on distance matrix
    * Differential abundance testing
    * Machine learning - Random forests
# Writing
  * What to report
  * Outlines!